Bot/functions.py

```diff
@@ -1,35 +1,173 @@
+    d = pytesseract.image_to_data(screenshot, output_type=pytesseract.Output.DICT)
+
+    # Initialize variables to store combined words and their coordinates
+    combined_words = []
+    current_word = ""
+    x_start, x_end, y_start, y_end = (
+        float("inf"),
+        float("-inf"),
+        float("inf"),
+        float("-inf"),
+    )
+
+    # Loop through detected text boxes
+    for i in range(len(d["level"])):
+        level = d["level"][i]
+        text = d["text"][i].strip()
+
+        # Only consider text boxes containing meaningful words
+        if level == 5 and text:
+            x, y, w, h = d["left"][i], d["top"][i], d["width"][i], d["height"][i]
+
+            # Check if the word is close to the previous word horizontally
+            if (
+                abs(x - x_end) <= CLOSE_WORDS_THRESHOLD
+            ):  # Adjust the threshold as needed
+                current_word += " " + text
+                x_end = max(x_end, x + w)
+                y_end = max(y_end, y + h)
+            else:
+                if current_word:
+                    combined_words.append(
+                        Sentence(
+                            current_word,
+                            x_start,
+                            y_start,
+                            x_end - x_start,
+                            y_end - y_start,
+                            (x_start + x_end) / 2,
+                            (y_start + y_end) / 2,
+                        )
+                    )
+                current_word = text
+                x_start, x_end = x, x + w
+                y_start, y_end = y, y + h
+
+    # Append the last word
+    if current_word:
+        combined_words.append(
+            Sentence(
+                current_word,
+                x_start,
+                y_start,
+                x_end - x_start,
+                y_end - y_start,
+                (x_start + x_end) / 2,
+                (y_start + y_end) / 2,
+            )
+        )
+
+    return combined_words
+
+
 # Functions
 ########################################################
 
 # Initialize a variable to store the last selected coordinate
-last_selected_coord: List[int] = []
+last_selected_coord: Sentence = Sentence("", 0, 0, 0, 0, 0.0, 0.0)
 
 
-def get_random_coord(coords: List[List[int]]) -> List[int]:
+def get_random_coord(coords: List[Sentence]) -> Sentence:
     global last_selected_coord
     available_coords = [coord for coord in coords if coord != last_selected_coord]
     selected_coord = random.choice(available_coords)
     last_selected_coord = selected_coord
     return selected_coord
 
 
-def auto_dock_to_station(coords: List[int]) -> None:
-    x, y = coords
-    click_top_left_circle_menu(x, y)
-    sleep_and_log(1)
-    click_top_center_circle_menu(x, y)
-    sleep_and_log(0.5)
+def auto_dock_to_station(x: int, y: int) -> None:
+    current_os = platform.system()
+
+    if current_os == "Darwin":  # macOS
```

Comment:
trying to find a way to support macOS . by right clicking and clicking the Dock option in the right click menu